src/api/ingestion.ts

```diff
@@ -21,16 +27,32 @@ function parsePdfIngestionResponse(response: IngestResponse): ReferenceItem[] {
   }));
 }
 
-export async function runPDFIngestion(): Promise<ReferenceItem[]> {
-  const uploadsDir = await getSystemPath(getUploadsDir());
-  const response = await callSidecar('ingest', { pdf_directory: String(uploadsDir) });
-  return parsePdfIngestionResponse(response);
+export async function runPDFIngestion(projectId?: string): Promise<ReferenceItem[]> {
+  if (import.meta.env.VITE_IS_WEB) {
+    if (!projectId) {
+      throw new Error('Project ID is required for web version');
```

Comment:
Couldn't we make the parameter mandatory instead? The frontend should not depend on the implementation and should send an id in any case, so it's not useful to make it optional, even if we don't use it in the sidecar implementation

Reply:
Yes. That will be implemented in the follow-up PR with the unified desktop+web code to use HTTP.